modules/common/service/service.go

```diff
@@ -70,6 +90,32 @@ func GenericService(svcInfo *GenericServiceDetails) *corev1.Service {
 	}
 }
 
+// MetalLBService func
+func MetalLBService(svcInfo *MetalLBServiceDetails) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        svcInfo.Name,
+			Namespace:   svcInfo.Namespace,
+			Annotations: svcInfo.Annotations,
+			Labels:      svcInfo.Labels,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: svcInfo.Selector,
+			Ports: []corev1.ServicePort{
+				{
+					Name: svcInfo.Port.Name,
+					Port: svcInfo.Port.Port,
+					// corev1.ProtocolTCP/ corev1.ProtocolUDP/ corev1.ProtocolSCTP
+					// - https://pkg.go.dev/k8s.io/api@v0.23.6/core/v1#Protocol
+					Protocol: svcInfo.Port.Protocol,
+				},
+			},
+			Type:           corev1.ServiceTypeLoadBalancer,
+			LoadBalancerIP: svcInfo.LoadBalancerIP,
```

Comment:
the `spec.LoadBalancerIP` is deprecated https://github.com/kubernetes/kubernetes/pull/107235 . we should switch to use the `metallb.universe.tf/loadBalancerIPs` instead. will update the PR